app/cdap/components/NamespaceAdmin/ServiceAccounts/EditConfirmDialog.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import T from 'i18n-react';
+import styled from 'styled-components';
+import TextField from '@material-ui/core/TextField';
+import { ConfirmDialog, SeverityType } from 'components/shared/ConfirmDialog';
+import {
+  validateServiceAccount,
+  addServiceAccount,
+} from 'components/NamespaceAdmin/store/ActionCreator';
+
+const PREFIX = 'features.ServiceAccounts';
+const GCLOUD_COMMAND_TEXT =
+  'gcloud iam service-accounts add-iam-policy-binding \\\n' +
+  '    --role roles/iam.workloadIdentityUser \\\n' +
+  '    --member "serviceAccount:${TENANT_PROJECT_ID}.svc.id.goog[default/${IDENTITY}]" \\\n' +
+  '    [GSA_EMAIL]';
+
+interface IEditConfirmDialogProps {
+  selectedServiceAcnt: any;
+  isShow: boolean;
+  closeFn: () => void;
+  setSelectedServiceAcnt: (arg0: any) => void;
+  isShowHelp?: boolean;
+  namespaceDetails: any;
+}
+
+const StyledTextField = styled(TextField)`
+  & .MuiInputLabel-shrink {
+    font-size: 17px;
+  }
+  & .MuiOutlinedInput-notchedOutline {
+    font-size: 16px;
+    border-color: #80868b;
+  }
+  & .MuiFormHelperText-root {
+    margin-left: 0px;
+    font-size: 12px;
+  }
+`;
+
+export const EditConfirmDialog = ({
```

Comment:
consider refactoring this component and avoid setting state inside useEffect hooks.